app/models/rad_consumer_session.rb

```diff
@@ -29,4 +31,22 @@ def store(firm_result, profile_url, most_recent_search_url)
   def firm_already_present?(firm_result)
     firms.any? { |firm_hash| firm_result.id == firm_hash['id'] }
   end
+
+  def most_recent_search_path_values(params)
```

Comment:
Could I suggest a rename to something like `locale_to_search_path_mappings` please?